### PR TITLE
ipq40xx: convert Mikrotik LHGG 60ad to DSA

### PR DESCRIPTION
**Pull request by PolynomialDivision**

### CODE REVIEW

**Comment by PolynomialDivision** [comment]
Run-tested (my etherner card on my pc is broken and is only doing 100 Mbit/s...^^):
```
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.10.113 (nick@cool) (arm-openwrt-linux-muslgnueabi-gcc (OpenWrt GCC 11.2.0 r19520+19-089eb02abc) 11.2.0, GNU ld (GNU Binutils) 2.37) #0 SMP Thu May 12 13:19:16 2022
[    0.000000] CPU: ARMv7 Processor [410fc075] revision 5 (ARMv7), cr=10c5387d
[    0.000000] CPU: div instructions available: patching division code
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: Mikrotik Wireless Wire Dish LHGG-60ad
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000080000000-0x000000008fffffff]
[    0.000000]   HighMem  empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000080000000-0x0000000087dfffff]
[    0.000000]   node   0: [mem 0x0000000087e00000-0x0000000087ffffff]
[    0.000000]   node   0: [mem 0x0000000088000000-0x000000008fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000080000000-0x000000008fffffff]
[    0.000000] On node 0 totalpages: 65536
[    0.000000]   Normal zone: 576 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 65536 pages, LIFO batch:15
[    0.000000] percpu: Embedded 15 pages/cpu s30732 r8192 d22516 u61440
[    0.000000] pcpu-alloc: s30732 r8192 d22516 u61440 alloc=15*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 64960
[    0.000000] Kernel command line: 
[    0.000000] Dentry cache hash table entries: 32768 (order: 5, 131072 bytes, linear)
[    0.000000] Inode-cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 237980K/262144K available (6271K kernel code, 603K rwdata, 1588K rodata, 10240K init, 234K bss, 24164K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] random: get_random_bytes called from start_kernel+0x360/0x50c with crng_init=0
[    0.000000] arch_timer: cp15 timer(s) running at 48.00MHz (virt).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0xb11fd3bfb, max_idle_ns: 440795203732 ns
[    0.000008] sched_clock: 56 bits at 48MHz, resolution 20ns, wraps every 4398046511096ns
[    0.000024] Switching to timer-based delay loop, resolution 20ns
[    0.000245] Calibrating delay loop (skipped), value calculated using timer frequency.. 96.00 BogoMIPS (lpj=480000)
[    0.000268] pid_max: default: 32768 minimum: 301
[    0.000457] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.000477] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.001551] CPU: Testing write buffer coherency: ok
[    0.001880] qcom_scm: convention: smc legacy
[    0.002671] Setting up static identity map for 0x80300000 - 0x8030003c
[    0.002810] rcu: Hierarchical SRCU implementation.
[    0.003020] dyndbg: Ignore empty _ddebug table in a CONFIG_DYNAMIC_DEBUG_CORE build
[    0.003295] smp: Bringing up secondary CPUs ...
[    0.006472] smp: Brought up 1 node, 4 CPUs
[    0.006500] SMP: Total of 4 processors activated (384.00 BogoMIPS).
[    0.006510] CPU: All CPU(s) started in SVC mode.
[    0.011405] VFP support v0.3: implementor 41 architecture 2 part 30 variant 7 rev 5
[    0.011582] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.011609] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.011900] pinctrl core: initialized pinctrl subsystem
[    0.013114] NET: Registered protocol family 16
[    0.013523] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.014667] thermal_sys: Registered thermal governor 'step_wise'
[    0.015101] cpuidle: using governor ladder
[    0.015160] cpuidle: using governor menu
[    0.036165] cryptd: max_cpu_qlen set to 1000
[    0.039824] usbcore: registered new interface driver usbfs
[    0.039902] usbcore: registered new interface driver hub
[    0.039968] usbcore: registered new device driver usb
[    0.040022] pps_core: LinuxPPS API ver. 1 registered
[    0.040108] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.040149] PTP clock support registered
[    0.041949] clocksource: Switched to clocksource arch_sys_counter
[    0.042909] NET: Registered protocol family 2
[    0.043080] IP idents hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.043732] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.043789] TCP established hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.043828] TCP bind hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.043879] TCP: Hash tables configured (established 2048 bind 2048)
[    0.044014] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.044057] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.044284] NET: Registered protocol family 1
[    0.044338] PCI: CLS 0 bytes, default 64
[    0.158879] workingset: timestamp_bits=14 max_order=16 bucket_order=2
[    0.163449] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.163477] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.296432] qcom-pcie 40000000.pci: host bridge /soc/pci@40000000 ranges:
[    0.296501] qcom-pcie 40000000.pci:       IO 0x0040200000..0x00402fffff -> 0x0040200000
[    0.296533] qcom-pcie 40000000.pci:      MEM 0x0040300000..0x0040ffffff -> 0x0040300000
[    0.573526] qcom-pcie 40000000.pci: Link up
[    0.573698] qcom-pcie 40000000.pci: PCI host bridge to bus 0000:00
[    0.573723] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.573742] pci_bus 0000:00: root bus resource [io  0x0000-0xfffff] (bus address [0x40200000-0x402fffff])
[    0.573758] pci_bus 0000:00: root bus resource [mem 0x40300000-0x40ffffff]
[    0.573809] pci 0000:00:00.0: [17cb:1001] type 01 class 0x060400
[    0.573840] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x00000fff]
[    0.573909] pci 0000:00:00.0: PME# supported from D0 D3hot
[    0.575420] PCI: bus0: Fast back to back transfers disabled
[    0.575703] pci 0000:01:00.0: [1ae9:0310] type 00 class 0x028000
[    0.575845] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
[    0.578007] PCI: bus1: Fast back to back transfers disabled
[    0.578066] pci 0000:00:00.0: BAR 8: assigned [mem 0x40400000-0x405fffff]
[    0.578085] pci 0000:00:00.0: BAR 0: assigned [mem 0x40300000-0x40300fff]
[    0.578110] pci 0000:01:00.0: BAR 0: assigned [mem 0x40400000-0x405fffff 64bit]
[    0.578182] pci 0000:00:00.0: PCI bridge to [bus 01-ff]
[    0.578201] pci 0000:00:00.0:   bridge window [mem 0x40400000-0x405fffff]
[    0.580995] pcieport 0000:00:00.0: AER: enabled with IRQ 68
[    0.583087] bam-dma-engine 8e04000.dma: num-channels unspecified in dt
[    0.583113] bam-dma-engine 8e04000.dma: num-ees unspecified in dt
[    0.583794] tcsr 1949000.tcsr: setting wifi_glb_cfg = 41000000
[    0.583900] tcsr 1953000.ess_tcsr: setting ess interface select = 1
[    0.583996] tcsr 1957000.tcsr: setting wifi_noc_memtype_m0_m2 = 2222222
[    0.584295] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.584899] msm_serial 78af000.serial: msm_serial: detected port #0
[    0.584948] msm_serial 78af000.serial: uartclk = 1843200
[    0.585002] 78af000.serial: ttyMSM0 at MMIO 0x78af000 (irq = 33, base_baud = 115200) is a MSM
[    0.585034] msm_serial: console setup on port #0
[    1.256590] printk: console [ttyMSM0] enabled
[    1.261767] msm_serial: driver initialized
[    1.270870] loop: module loaded
[    1.272025] spi_qup 78b5000.spi: IN:block:16, fifo:64, OUT:block:16, fifo:64
[    1.274871] spi-nor spi0.0: w25q128jv (16384 Kbytes)
[    1.280170] 3 fixed-partitions partitions found on MTD device spi0.0
[    1.285121] Creating 3 MTD partitions on "spi0.0":
[    1.291410] 0x000000000000-0x000000080000 : "Qualcomm"
[    1.296651] 0x000000080000-0x000000100000 : "RouterBoot"
[    1.308986] 3 routerbootpart partitions found on MTD device RouterBoot
[    1.309022] Creating 3 MTD partitions on "RouterBoot":
[    1.314460] 0x00000000e000-0x000000010000 : "hard_config"
[    1.320096] 0x000000010000-0x000000017c31 : "dtb_config"
[    1.325552] 0x00000003d000-0x00000003e000 : "soft_config"
[    1.330888] 0x000000100000-0x000001000000 : "firmware"
[    1.339254] random: fast init done
[    1.343960] 2 minor-fw partitions found on MTD device firmware
[    1.344098] Creating 2 MTD partitions on "firmware":
[    1.349918] 0x000000000000-0x000000300000 : "kernel"
[    1.355535] 0x000000300000-0x000000f00000 : "rootfs"
[    1.360520] mtd: device 7 (rootfs) set to be root filesystem
[    1.365201] 1 squashfs-split partitions found on MTD device rootfs
[    1.370583] 0x000000620000-0x000000f00000 : "rootfs_data"
[    1.394832] ipqess-edma c080000.ethernet: generated random MAC address 9e:fd:3c:10:e2:51
[    1.418248] i2c /dev entries driver
[    1.420088] sdhci: Secure Digital Host Controller Interface driver
[    1.420549] sdhci: Copyright(c) Pierre Ossman
[    1.426835] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.437605] MikroTik RouterBOARD hardware configuration sysfs driver v0.07
[    1.439549] MikroTik RouterBOARD software configuration sysfs driver v0.05
[    1.445005] NET: Registered protocol family 10
[    1.452266] Segment Routing with IPv6
[    1.455047] NET: Registered protocol family 17
[    1.458962] 8021q: 802.1Q VLAN Support v1.8
[    1.463096] Registering SWP/SWPB emulation handler
[    1.501851] qca8k-ipq4019 c000000.switch: configuring for fixed/internal link mode
[    1.502226] qca8k-ipq4019 c000000.switch: Link is Up - 1Gbps/Full - flow control rx/tx
[    1.585923] qca8k-ipq4019 c000000.switch lan (uninitialized): PHY [90000.mdio-1:00] driver [Qualcomm Atheros AR8035] (irq=POLL)
[    1.598182] DSA: tree 0 setup
[    1.627254] Freeing unused kernel memory: 10240K
[    1.645298] Run /init as init process
[    1.645528]   with arguments:
[    1.645608]     /init
[    1.645675]   with environment:
[    1.645747]     HOME=/
[    1.645816]     TERM=linux
[    2.010589] init: Console is alive
[    2.010919] init: - watchdog -
[    2.023655] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    2.027728] genirq: irq_chip msmgpio did not update eff. affinity mask of irq 69
[    2.044143] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    2.054229] init: - preinit -
[    2.340130] random: jshn: uninitialized urandom read (4 bytes read)
[    2.393795] random: jshn: uninitialized urandom read (4 bytes read)
[    2.436419] random: jshn: uninitialized urandom read (4 bytes read)
[    2.692314] ipqess-edma c080000.ethernet eth0: configuring for fixed/internal link mode
[    2.693221] qca8k-ipq4019 c000000.switch lan: configuring for phy/rgmii-id link mode
[    2.699296] ipqess-edma c080000.ethernet eth0: Link is Up - 1Gbps/Full - flow control rx/tx
[    2.702202] 8021q: adding VLAN 0 to HW filter on device lan
[    2.715583] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[    6.941588] procd: - early -
[    6.941868] procd: - watchdog -
[    7.490427] procd: - watchdog -
[    7.495186] procd: - ubus -
[    7.555758] procd: - init -
[    7.951162] kmodloader: loading kernel modules from /etc/modules.d/*
[    7.957617] Loading modules backported from Linux version v5.15.33-0-g06f50ca83ace
[    7.957665] Backport generated by backports.git v5.15.33-1-0-g183c4ab2
[    7.985591] urngd: v1.0.2 started.
[    8.031448] PPP generic driver version 2.4.2
[    8.033869] NET: Registered protocol family 24
[    8.042690] wil6210 0000:01:00.0: wil6210 device found [1ae9:0310] (rev 2) bar size 0x200000
[    8.043248] wil6210 0000:01:00.0: enabling device (0140 -> 0142)
[    8.050334] wil6210 0000:01:00.0 (unnamed net_device) (uninitialized): wil_pcie_probe: CSR at [mem 0x40400000-0x405fffff 64bit] -> 0x(ptrval)
[    8.056318] wil6210 0000:01:00.0 (unnamed net_device) (uninitialized): wil_set_capabilities: Board hardware is Sparrow B0, flash exist
[    8.068824] wil6210 0000:01:00.0 (unnamed net_device) (uninitialized): wil_set_capabilities: platform_capa 0x0
[    8.072029] random: crng init done
[    8.090824] random: 7 urandom warning(s) missed due to ratelimiting
[    8.092628] wil6210 0000:01:00.0 (unnamed net_device) (uninitialized): wil_refresh_fw_capabilities: keep_radio_on_during_sleep (0)
[    8.100313] wil6210 0000:01:00.0: using dma mask 48
[    8.192037] wil6210 0000:01:00.0 (unnamed net_device) (uninitialized): wil_get_bl_info: Boot Loader struct v2: MAC = 08:c5:e1:da:6b:5b RF = 0x0000 (status 0x0000) bband = 0x00000000
[    8.192102] wil6210 0000:01:00.0 (unnamed net_device) (uninitialized): wil_get_bl_info: Boot Loader build 255.255.0.6836
[    8.207224] wil6210 0000:01:00.0 (unnamed net_device) (uninitialized): wil_set_oob_mode: oob_mode to 0
[    8.227872] kmodloader: done loading kernel modules from /etc/modules.d/*
[   23.772776] ipqess-edma c080000.ethernet eth0: Link is Down
[   23.780281] ipqess-edma c080000.ethernet eth0: configuring for fixed/internal link mode
[   23.780624] ipqess-edma c080000.ethernet eth0: Link is Up - 1Gbps/Full - flow control rx/tx
[   23.784071] device eth0 entered promiscuous mode
[   23.795601] qca8k-ipq4019 c000000.switch lan: configuring for phy/rgmii-id link mode
[   23.800940] 8021q: adding VLAN 0 to HW filter on device lan
[   23.808562] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   23.814322] br-lan: port 1(lan) entered blocking state
[   23.819606] br-lan: port 1(lan) entered disabled state
[   23.825536] device lan entered promiscuous mode
[   94.554754] qca8k-ipq4019 c000000.switch lan: Link is Up - 100Mbps/Full - flow control off
[   94.555314] br-lan: port 1(lan) entered blocking state
[   94.562648] br-lan: port 1(lan) entered forwarding state
[   94.574129] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
```

**Comment by PolynomialDivision** [comment]
Works perfectly, please merge @robimarko

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-lhgg-60ad.dts [hunk 1]
```diff
@@ -47,6 +47,10 @@
 
 		mdio@90000 {
 			status = "okay";
+
+			ethphy: ethernet-phy@0 {
+				sreg = <0x0>;
```
Comment (robimarko) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-lhgg-60ad.dts:52]: This is not correct.
It's `reg` and not `sreg`.

There is no need for a new PHY node, just reuse the ethphy0 one.

**Comment by robimarko** [comment]
I pushed the version with ethphy reused, can you please test and report back.

**Pull request closed without merging**